ml-agents/mlagents/trainers/tests/test_simple_rl.py

```diff
@@ -262,28 +182,45 @@ def _check_environment_trains(
 
         # Begin training
         tc.start_learning(env_manager)
-        print(tc._get_measure_vals())
         if (
             success_threshold is not None
         ):  # For tests where we are just checking setup and not reward
-            for mean_reward in tc._get_measure_vals().values():
-                assert not math.isnan(mean_reward)
-                assert mean_reward > success_threshold
+
+            processed_rewards = [
```

Comment:
Nice, this seems a lot cleaner than the old `_get_measure_vals()` approach.

We might be able to remove or refactor `_get_measure_vals()` now too.

Reply:
I think the `_get_measure_vals()` is still used by the MetaCurriculum to increment the lessons. It's a bit weird since it's the only reason the Trainer keeps track of rewards at all; for most reporting it happens in the EnvManager's AgentProcessors. 
Instead we can have the `_get_measure_vals()` or equivalent grab the rewards from the EnvManager, this should be pretty easy